### PR TITLE
Fix figure window icon

### LINKED ISSUE #11138: Only the first figure window has mpl icon, all other figures have default tk icon.
### Bug report

**Bug summary**
When plot more than one figure window, only the first figure window has the default mpl icon (come from matplotlib.ppm), and all other figure windows' icon is set to the default tk icon.

**Code for reproduction**

code 1:
```python
import logging
import matplotlib.pyplot as plt
logging.basicConfig(format="%(name)s %(levelname)s: %(message)s", level=logging.INFO)
plt.figure(1)
plt.plot([1,2,3])
plt.show()
```

code 2:
```python
import logging
import matplotlib.pyplot as plt
logging.basicConfig(format="%(name)s %(levelname)s: %(message)s", level=logging.INFO)
plt.figure(1)
plt.plot([1,2,3])
plt.figure(2)
plt.plot([4,5,6])
plt.show()
```

code 3:
```python
import logging
import matplotlib.pyplot as plt
logging.basicConfig(format="%(name)s %(levelname)s: %(message)s", level=logging.INFO)
plt.figure(1)
plt.plot([1,2,3])
plt.figure(2)
plt.plot([4,5,6])
plt.figure(3)
plt.plot([7,8,9])
plt.show()
```

**Actual outcome**
code 1 runs normally.

code 2 will log:
```
matplotlib.backends._backend_tk INFO: Could not load matplotlib icon: can't use "pyimage10" as iconphoto: not a photo image
```
And the icon of figure 1 and figure 2 is different.

code 3 will log:
```
matplotlib.backends._backend_tk INFO: Could not load matplotlib icon: can't use "pyimage10" as iconphoto: not a photo image
matplotlib.backends._backend_tk INFO: Could not load matplotlib icon: can't use "pyimage19" as iconphoto: not a photo image
```
The icon of figure 1 is different than figure 2 and 3.
Screenshot attached:
Icon of window:
figure_title

And the icon in taskbar:
taskbar

If debug, the log information is coming from the exception:
```
file: matplotlib/backends/_backend_tk.py
class _BackendTk()
method new_figure_manager_given_figure()
        # Put a mpl icon on the window rather than the default tk icon.
        # Tkinter doesn't allow colour icons on linux systems, but tk>=8.5 has
        # a iconphoto command which we call directly. Source:
        # http://mail.python.org/pipermail/tkinter-discuss/2006-November/000954.html
        icon_fname = os.path.join(
            rcParams['datapath'], 'images', 'matplotlib.ppm')
        icon_img = Tk.PhotoImage(file=icon_fname)
        try:
            window.tk.call('wm', 'iconphoto', window._w, icon_img)
        except Exception as exc:
            # log the failure (due e.g. to Tk version), but carry on
            _log.info('Could not load matplotlib icon: %s', exc)
```

**Expected outcome**
All figure's icon are same by default.

**Matplotlib version**
  * Operating system: Windows 7 pro.
  * Matplotlib version:  2.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): TkAgg
  * Python version: 3.6.5

### PR DESCRIPTION
**Pull request by souravsingh**
## PR Summary
Fixes #11138 

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
Thanks a lot for this and youf first contribution!

Does this fix the problem?  PRs are more likely to get a quick review if you don't ask us to download and test by hand, particularly for GUI interaction issues. Also a little more detail than a bare link to the issue is nice, and a bit of description as to why this is an improvement.   To show this fixes the issue, a screenshot should suffice...

**Comment by jklymak** [comment]
ping!

**Comment by souravsingh** [comment]
I have rebased the PR. Let me test the PR out

File: lib/matplotlib/backends/_backend_tk.py [hunk 1]
```diff
@@ -1004,29 +1004,28 @@ def new_figure_manager_given_figure(cls, num, figure):
...
-                # log the failure (due e.g. to Tk version), but carry on
-                _log.info('Could not load matplotlib icon: %s', exc)
-
-            canvas = cls.FigureCanvas(figure, master=window)
-            manager = cls.FigureManager(canvas, num, window)
-            if matplotlib.is_interactive():
-                manager.show()
-                canvas.draw_idle()
-            return manager
+        window = Tk.Tk(className="matplotlib")
```
Comment (anntzer) [inline lib/matplotlib/backends/_backend_tk.py]: The rebase incorrectly got rid of the contextmanager.
Anyone can dismiss this review once fixed.

**Comment by tacaswell** [comment]
I tested that this fixes the problem locally.

**Comment by tacaswell** [comment]
Thanks @souravsingh !

**Pull request merged**